File: src/my_py_pkg/my_py_pkg/add_two_ints_client.py
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

class AddTwoIntClientNode(Node):
    def __init__(self):
        super().__init__("add_two_ints_client")
        self.client = self.create_client(AddTwoInts, "add_two_ints")
        self.call_service(3, 8)

    def call_service(self, a: int, b:int):
        while not self.client.wait_for_service(1.0):
            self.get_logger().warn("waiting for Add Two Ints server ...")

        request = AddTwoInts.Request(a=a, b=b)
        future = self.client.call_async(request)
        future.add_done_callback(self.callback_response)

    def callback_response(self, future):
        response = future.result()
        self.get_logger().info(f"Received: {response.sum}")


def main(args=None):
    rclpy.init(args=args)
    rclpy.spin(AddTwoIntClientNode())
    rclpy.shutdown()

File: src/my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"

#include "example_interfaces/srv/add_two_ints.hpp"

class AddTwoIntsClientNode : public rclcpp::Node {
public:
    AddTwoIntsClientNode() : rclcpp::Node("add_two_ints_client") {
        client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
        this->addTwoInts(6, 9);
    }

private:
    void addTwoInts(int64_t a, int64_t b) {
        while (not client->wait_for_service(std::chrono::seconds(1))) {
            RCLCPP_WARN(this->get_logger(), "waiting for the server ...");
        }

        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a = a;
        request->b = b;

        auto future = client
            ->async_send_request(
                request,
                std::bind(&AddTwoIntsClientNode::callbackAddTwoIntsRequested, this, std::placeholders::_1)
            );
    }

    void callbackAddTwoIntsRequested(rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future) {
        auto response = future.get();
        RCLCPP_INFO(this->get_logger(), "Sum: %ld", response->sum);
    }


    rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client;
};


int main(int argc, const char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<AddTwoIntsClientNode>());
    rclcpp::shutdown();
    
    return 0;
}